contracts/RadiCards.sol

```diff
@@ -3,234 +3,504 @@ pragma solidity ^0.4.24;
+            daiDonation: false,
+            giftAmount : _giftAmount,
+            donationAmount: _donationAmount,
+            status: _giftStatus,
+            cardIndex : _cardIndex,
+            benefactorIndex : _benefactorIndex
+        });
+
+        // Card is minted to the _sentToAddress. This is either this radicards contract(if claimableLink==true)
+        // and the creator chose to use the escrow for a claimable link or to the recipient EOA directly
+        uint256 _tokenId = _mint(_sentToAddress, cards[_cardIndex].tokenURI);
+        cards[_cardIndex].minted++;
+
+        // transfer the ETH to the benefactor
+        if(_donationAmount > 0){
+            benefactors[_benefactorIndex].ethAddress.transfer(_donationAmount);
+        }
+        // transfer gift to recipient.
+        // note that we only do this if the link is not claimable as if it is the eth sits in escrow within this contract
+        if(_giftAmount > 0 && !_claimableLink){
+            _sentToAddress.transfer(_giftAmount);
+        }
+
+        // tally up the total eth gifted and donated
+        totalGiftedInWei = totalGiftedInWei.add(_giftAmount);
+        totalDonatedInWei = totalDonatedInWei.add(_donationAmount);
+
+        emit CardGifted(_sentToAddress, _benefactorIndex, _cardIndex, msg.sender, _tokenId, _giftStatus);
+        return true;
+    }
+
+    function giftInDai(address to, uint256 _benefactorIndex, uint256 _cardIndex, string _message, uint256 _donationAmount, uint256 _giftAmount, bool _claimableLink) public payable returns (bool) {
+        require(to != address(0), "Must be a valid address");
+        require(benefactors[_benefactorIndex].ethAddress != address(0), "Must specify existing benefactor");
+        require(bytes(cards[_cardIndex].tokenURI).length != 0, "Must specify existing card");
+        require(cards[_cardIndex].active, "Must be an active card");
+        require((_donationAmount + _giftAmount)<= daiContract.allowance(msg.sender, this), "Must have provided high enough alowance to Radicard contract");
+        require((_donationAmount + _giftAmount)<= daiContract.balanceOf(msg.sender), "Must have enough token balance of dai to pay for donation and gift amount");
+
+        if (cards[_cardIndex].maxQnty > 0){ //the max quantity is set to zero to indicate no limit. Only need to check that can mint if limited
+            require(cards[_cardIndex].minted < cards[_cardIndex].maxQnty, "Can't exceed maximum quantity of card type");
+        }
+
+        if(cards[_cardIndex].minPrice > 0){ //if the card has a minimum price check that enough has been sent
+            require((_donationAmount + _giftAmount) >= cards[_cardIndex].minPrice, "The total dai sent with the transaction is less than the min price of the token");
+        }
+
+        //parameters that change based on the if the card is setup as with a claimable link
+        Statuses _giftStatus;
+        address _sentToAddress;
+
+        if(_claimableLink){
```

Comment:
Not 100% sure of my understanding of the claimable link stuff, but does this mean that it sends them  `DAI` of `EPHEMERAL_ADDRESS_FEE` amount and not ETH - is this correct and as designed?

Reply:
nope. this should send the ephemeral address the `EPHEMERAL_ADDRESS_FEE ` in ETH as its calling the .transfer function on the address. this is verified in the unit tests